### DISCUSSION TITLE
Is there a way to set supports_statement_cache to True when creating create_engine?

### DISCUSSION THREAD

**Discussion by fabricebaranski** [sqlalchemy/sqlalchemy#8841] [Usage Questions]
I am using Modin to get data from a database using ModinDatabaseConnection.
As kwargs are passed directly to create_engine of sqlalchemy, I would like to know if it exits a possibility to set supports_statement_cache to True in parameters of create_engine function?

**Comment by CaselIT** [comment]
Hi,

Statement caching is enabled by default, so there should not be any need to manually enable it.
Can I ask what's the source of the question?

**Reply by zzzeek** [reply]
this is the per-dialect attribute that defaults to False, so that third party dialects weren't surprised into supporting caching.    some docs on some edge cases I found are at https://docs.sqlalchemy.org/en/14/core/connections.html#caching-for-third-party-dialects , these are edge cases where caching would cause query failures if the dialect itself weren't vetted for these cases.

if you want to *try* caching with a third party dialect, do `e = create_engine(...)`, then set the attribute as `e.dialect.supports_statement_cache = True`.  That's not a long term solution however, the correct solution is contact the maintainers of this DBAPI and have them test their dialect with caching in 1.4.

**Comment by fabricebaranski** [comment]
In fact, the datasource is Druid (using pydruid).
As I have no way to set supports_statement_cache to True on each ray's worker (Modin is above Ray), I wonder if one of args in create_engine can concern DruidDialect.

**Reply by CaselIT** [reply]
No, since it's not something that can be turned on with a flag. You would need to ask the dialect developed to support it
